src/vulkan-renderer/imgui.cpp

```diff
@@ -83,26 +83,44 @@ ImGUIOverlay::ImGUIOverlay(const wrapper::Device &device, const wrapper::Swapcha
     m_push_const_block.translate = glm::vec2(0.0f, 0.0f);
 
     // Load font texture
-    unsigned char *font_texture_data{};
+    unsigned char* font_texture_data{};
+    unsigned char* font_error_texture_data{};
     int font_texture_width{0};
     int font_texture_height{0};
+    int font_texture_channels{4};
 
     const char *font_filename = "assets/fonts/NotoSans-Bold.ttf";
 
     spdlog::debug("Loading front '{}'", font_filename);
 
-    io.Fonts->AddFontFromFileTTF(font_filename, 18.0f);
-
-    io.Fonts->GetTexDataAsRGBA32(&font_texture_data, &font_texture_width, &font_texture_height);
-    VkDeviceSize uploadSize = font_texture_width * font_texture_height * 4 * sizeof(char);
+    ImFont* font = io.Fonts->AddFontFromFileTTF(font_filename, 18.0f);
+    if (font == nullptr) {
+        spdlog::error("Unable to load font {}.  Falling back to error texture.", font_filename);
+        font_error_texture_data = wrapper::Texture::load_error_texture_data(&font_texture_width, &font_texture_height, &font_texture_channels);
+        assert(font_error_texture_data);
+        font_texture_data = font_error_texture_data;
+    } else {
+        io.Fonts->GetTexDataAsRGBA32(&font_texture_data, &font_texture_width, &font_texture_height);
+        if (font_texture_data == nullptr) {
```

Comment:
If you want to test the code for this change, you can flip this to inequality and run it and it will draw the error texture.

Before merging, I will probably revert this entire file back as I do not think these changes are really meaningful outside of debugging and testing the error texture.

If a font fails to load, the texture information will only be part of what is missing.  The UI will also be missing all the sizing and measurement data from the font.  The UI is basically doomed without a font.

So, anyways, I just wanted to leave this in for the initial review in case anyone wanted to see it or test it.  But, for the final review, I probably need to revert this file.

Reply:
Also, if we revert this for the merge, I probably need to make the "load_error_texture_data" function a private member instead of a public static.